syntax = "proto3";

package genx.llm_loader;

service LLMLoader {
  rpc LoadModel(LoadModelRequest) returns (LoadModelResponse);
  rpc UnloadModel(UnloadModelRequest) returns (UnloadModelResponse);
  rpc GetModelDetails(GetModelDetailsRequest) returns (GetModelDetailsResponse);
}

message LoadModelRequest {
  string controller = 1; // "tgi" or "vllm"
  string model_id = 2;
  string quantization_type = 3; // For TGI, e.g., "gptq"
  map<string, string> parameters = 4; // Additional model params
  string device = 5; // "cpu" or "gpu"
}

message LoadModelResponse {
  bool success = 1;
  string message = 2;
  string model_id = 3;
}

message UnloadModelRequest {
  string model_id = 1;
}

message UnloadModelResponse {
  bool success = 1;
  string message = 2;
}

message GetModelDetailsRequest {
  string model_id = 1;
}

message GetModelDetailsResponse {
  bool success = 1;
  string model_id = 2;
  string controller = 3;
  string device = 4;
  map<string, string> parameters = 5;
  string message = 6;
}